marathon/resource_marathon_app.go

```diff
@@ -1163,6 +1237,11 @@ func mutateResourceToApplication(d *schema.ResourceData) *marathon.Application {
 				}
 			}
 
+			if prop, ok := mapStruct["port"]; ok {
+				prop := prop.(int)
+				healthCheck.Port = &prop
```

Comment:
This will always set port to 0, which is wrong in case you're using portIndex.

I'd suggest to add an conditional to check if it's greater than 0.

```
			if prop, ok := mapStruct["port"]; ok {
				prop := prop.(int)
				if prop > 0 {
					healthCheck.Port = &prop
				}
			}
```